File: src/main/java/cz/stasimek/fakturaceeasypeasy/enumeration/SubjectLegalForm.java
package cz.stasimek.fakturaceeasypeasy.enumeration;

import java.util.Arrays;
import java.util.LinkedHashMap;
import java.util.Map;
import java.util.stream.Collectors;
import lombok.Getter;

@Getter
public enum SubjectLegalForm {
	INDIVIDUAL("Fyzická osoba"),
	LEGAL_ENTITY("Právnická osoba");

	private final String nameCs;

	SubjectLegalForm(String nameCs) {
		this.nameCs = nameCs;
	}

	public static Map<String, String> valuesMap() {
		return Arrays.stream(SubjectLegalForm.values())
				.map((SubjectLegalForm s) -> new String[]{s.name(), s.getNameCs()})
				.collect(Collectors.toMap(p -> p[0], p -> p[1], (x, y) -> y, LinkedHashMap::new));
	}
}

File: src/main/kotlin/cz/stasimek/fakturaceeasypeasy/enumeration/SubjectLegalForm.kt
package cz.stasimek.fakturaceeasypeasy.enumeration

enum class SubjectLegalForm(private val nameCs: String) {
    INDIVIDUAL("Fyzická osoba"),
    LEGAL_ENTITY("Právnická osoba");

    companion object {
        fun valuesMap(): Map<String, String> {
            return values()
                .map { it.name to it.nameCs }
                .toMap(LinkedHashMap())
        }
    }
}